lib/services/access_request.go

```diff
@@ -2582,6 +2610,68 @@ func getKubeResourcesFromResourceIDs(resourceIDs []types.ResourceID, clusterName
 	return kubernetesResources, nil
 }
 
+// canEventuallyAccessSAMLApp checks access to saml resource with a [RoleSet]
+// created from requested and standing roles.
+// AccessDenied error is returned as a BadParameter to indicate error is
+// related to the access request configuration.
+func (m *RequestValidator) canEventuallyAccessSAMLApp(ctx context.Context, requestedRoles []string, requestedResourceIDs []types.ResourceID) error {
+	if len(requestedResourceIDs) == 0 {
+		return nil
+	}
+
+	var samlResourceIDs []types.ResourceID
+	for _, r := range requestedResourceIDs {
+		if r.Kind == types.KindSAMLIdPServiceProvider {
+			samlResourceIDs = append(samlResourceIDs, r)
+		}
+	}
+	if len(samlResourceIDs) == 0 {
+		return nil
+	}
+
+	resources, err := m.getUnderlyingResourcesByResourceIDs(ctx, samlResourceIDs)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	allRoles := slices.Concat(requestedRoles, m.userState.GetRoles())
+	roleSet, err := FetchRoles(allRoles, m.getter, m.userState.GetTraits())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	cluster, err := m.getter.GetClusterName(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	for _, r := range resources {
+		if r.GetKind() != types.KindSAMLIdPServiceProvider {
+			continue
+		}
+
+		if err := roleSet.CheckAccessToSAMLIdP(
+			r,
+			m.userState.GetTraits(),
+			nil, /*AuthPreference, checked during login*/
+			AccessState{MFAVerified: true},
+		); err != nil {
+			if trace.IsAccessDenied(err) {
+				return trace.BadParameter(
+					`no roles configured in the "search_as_roles" for this user allow `+
```

Comment:
How a user can issue a JIT Access Request with some requestedResourceIDs  set and then get the errro that he don't have the `search_as_roles` ? 


If user don't have any role `search_as_roles` my thinking is that we should not be able to issue any JIT access request because  it don't have permission to list any resources.